src/v/archival/upload_housekeeping_service.cc

```diff
@@ -399,19 +399,18 @@ ss::future<> housekeeping_workflow::run_jobs_bg() {
                 jobs_executed++;
                 quota = res.remaining;
                 maybe_update_probe(res);
-            } catch (const ss::gate_closed_exception&) {
-                // Shutting down
-            } catch (const ss::abort_requested_exception&) {
-                // Shutting down
             } catch (...) {
-                vlog(
-                  archival_log.error,
-                  "upload housekeeping job {} error: {}",
-                  job_name,
-                  std::current_exception());
-                jobs_failed++;
-                maybe_update_probe(
-                  {.status = housekeeping_job::run_status::failed});
+                auto eptr = std::current_exception();
+                if (!ssx::is_shutdown_exception(eptr)) {
```

Comment:
I'd still log the other exceptions at trace level after looking at the implementation of `is_shutdown_exception`. Broken promise is probably fine to ignore, but only if we are really shutting down. But what if it's thrown in other cases? I.e. a broken promise thrown due to a bug where the original promise went out of scope. I've had one of this where in one edge case I didn't return a correct value. Good luck debugging it if you're hiding it here. And, to be honest, in all other places where `is_shutdown_exception` is used :)

Reply:
I have had similar concerns blanket ignoring some of these exceptions. The pattern is pretty deep in our codebase at this point

Reply:
Yeh agree. @andrwng are you tracking this? Thanks for reviewing folks even after the merge.

Reply:
Yeah it's a good point. Will update this

Reply:
Put up https://github.com/redpanda-data/redpanda/pull/14986 to add a debug log message here and a couple other places